### ISSUE TITLE
[UX] Layout listing: Add help text, to explain how the order and grouping in that page works

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4868]
This is related to #4841 and was brought up during the Dec. 31, 2020 weekly UX meeting, https://youtu.be/Fc-oQ55LQi0?t=2190 (~37min into the call)

@philsward (referring to the layout listing page) explained:

> ...just coming and looking at it, I feel that everything is jumbled together ...I don't know what I'm looking at. Nothing is organized, and that was kinda my big beaf with it.

@jenlampton said:

> ...everything _is_ organized, but it's not clear to anyone looking at the page that it is. ...it's confusing.

Then @stpaultim asked a question about the organization of things:

> ...I've always understood that the higher up a layout is on the list, the more priority it has.

...then @jenlampton went on, to explain how the grouping and priority works on the layout listing page.

@cellear:

> I just learned 30sec ago that the order in this page matters. I had no idea.

@stpaultim:

> It doesn't say that anywhere. ...the only reason I think I know that is I think I've heard @jenlampton say it, but I might have misunderstood.

@jenlampton 

> ...and if you have 2 things that _can_ be ordered, a button shows up that says "Reorder". But if you don't ever have them, you would never know that. ...if you never create anything that needs an order, you'll never even know that there _are_ things that can have an order.

This is what @jenlampton was referring to:

![Screen Shot 2021-01-12 at 6 04 08 pm](https://user-images.githubusercontent.com/2423362/104339812-b03bc480-5500-11eb-8903-07214e98af85.png)

---

## Proposed solution:

<img width="1278" alt="Screen Shot 2022-01-03 at 2 37 35 PM" src="https://user-images.githubusercontent.com/397895/147987923-2d4b8e44-de81-45f8-bdb1-893c906ce027.png">

**Comment by klonos** [comment]
PS: @stpaultim mentioned that when creating a node-specific layout (like `node/3`), he noticed that it is listed _after_ the "catch-all" `node/%` layout, but `node/3` takes precedence over `node/%`. This will need testing/confirmation, to clarify if it is expected behavior or a bug. If a bug, we'll need a follow-up issue for it.

**Comment by klonos** [comment]
Suggested help text (broken down to short sentences, in the form of bullet points):

> - The various paths that layouts handle are listed here alphabetically.
> - Paths not listed here have no layouts explicitly defined for them. In that case, one of the _Default layouts_ will be used (listed at the bottom of the table).
> - Layouts are grouped by the path they handle.
> - When looking to pick a layout to be used for a page, a single path from this table is matched.
> - Each path listed here may have either one or multiple layouts defined within its group.
> - If any path has...
>   - **exactly one layout defined**: then that layout will be used (unless visibility conditions rule it out).
>   - **multiple layouts defined**: starting from top to bottom, all layouts within the path group are checked for matching visibility conditions. The first layout within the path group that matches all conditions will be the one that gets used. If no match is found, it will fall back to one of the Default layouts.
> For more information and details, see the [online documentation]().

**Comment by klonos** [comment]
Although people new to layouts may appreciate this help text, adding such a big chunk of text in that already busy page may be annoying to people that already know how layouts work. Also, it may become less useful after people have learned how things work.

For some time, I have liked how Wordpress handles "inline" help, and wanted to propose that we do something similar to this:

![wp-help-expand-collapse](https://user-images.githubusercontent.com/2423362/104359660-9c9c5800-5518-11eb-9550-e3557a4f515b.gif)

...well, I think that this issue here is the perfect opportunity to test and consider a new UI pattern for such situations. So here's a PoC PR, which uses a collapsible, and collapsed-by-default fieldset, to mimic the WP expanding/collapsing help widget: https://github.com/backdrop/backdrop/pull/3492

![backdrop-help-expand-collapse](https://user-images.githubusercontent.com/2423362/104359687-a3c36600-5518-11eb-9e53-4a8fdd0c8aa3.gif)

**Comment by philsward** [comment]
The suggestion looks good and I like the idea of site wide help. I think it's worth exploring.

I keep thinking I will get around to a mock-up of how I think the GUI could be easier to understand but haven't had the time to dedicate to it. 😕

**Comment by stpaultim** [comment]
I love this idea of contextual help. My only concern is that we have many other pages that could benefit from this kind of help text and is this the pattern we want to adopt or is there a better one. We've also discussed tooltip solutions. 

https://github.com/backdrop/backdrop-issues/issues/3707

One of the blockers for any kind of "help" module in core seems to be the lack of content for it and uncertainly that such content would ever get created if the module existed. Chicken vs Egg dilemma. 

Maybe we could adopt some UI/Design scheme like this as a placeholder for a larger "help" module in the future, with the idea that anything put into this kind of accordion styled help block would be moved to any new help module in the future.

Just a thought.

**Comment by klonos** [comment]
> We've also discussed tooltip solutions.

I know, but the solution I've used in this PoC is basically re-using something that exists in core already (simple fieldsets with text of `'#type' => 'help'`), whereas something like #3707 would require much more work (and most likely introducing another 3rd party library in core).

> Maybe we could adopt some UI/Design scheme like this as a placeholder for a larger "help" module in the future, with the idea that anything put into this kind of accordion styled help block would be moved to any new help module in the future.

Yeah, the `help` module was removed from Backdrop core. See https://github.com/backdrop/backdrop-issues/issues/75#issuecomment-43466880

> Yep, it's expected that you'll add help on the pages directly, such as $form['help']['#markup'] = $help. Although I think it'd be a good idea for us to make some kind of standard help wrapper so you don't have to be responsible for wrapping paragraphs tags and classes.

**Comment by bugfolder** [comment]
Looking at both the expandable-help (described in this issue) and the tooltip suggestion (in #3707), while both would be big improvements, I favor the tooltip approach for two reasons.

1. It's smaller in the collapsed state. The expanding-help fieldset takes up a full "row" of space, even when collapsed. The tooltip is just another character, and so it can be tucked into places that a full row would be problematic. That allows tooltip-help to be more fine-grained; you can put lots of different tooltips in a form (or even on individual items in a table), so that individual tooltips can be more precisely targeted.

2. This is just a personal aesthetic, but having a popup overlay appear (with a tooltip) seems a little bit friendlier than having something expand and push page content down and out of the way. I recognize that having the extra information push down the existing page content gives the user the option to scroll back and forth between the help text and what it applies to, while an overlap covers up what it applies to, so I see arguments on both sides. On interfaces that support hovering (e.g., desktop, vs mobile), though, it would be nice to have the help text appear on hover (or let that be a theming option).

@stpaultim pointed out that an issue would be the lack of existing content, which is true. If tooltips started appearing widely in core, though, that would be an incentive for contribs to adopt it. And if adding a tooltip was as simple as adding the `'#tooltip' => ...` to form elements and render arrays, creating tooltips in both core and contrib would be easy ways for the community to contribute. One of the intellectual barriers to submitting one's first PR is the concern "does my solution have a bunch of side effects that I don't recognize or run afoul of practices I don't know about yet?" Creating tooltip PRs would be "good first issues".

**Comment by bugfolder** [comment]
Based on PR https://github.com/backdrop/backdrop/pull/3492, I've submitted PR https://github.com/backdrop/backdrop/pull/3754 with these differences:

* Use the new `details` element instead of a fieldset. (H/T @klonos )
* Some changes in wording, in particular noting difference between system path and aliases.
* Also changed wording under the assumption that the formatting changes in PR https://github.com/backdrop/backdrop/pull/3732 (separating layout listing into "Layout pages" and "Layout overrides") will be applied as well.
* The link to the "online documentation" previously just went to a listing of menu items. I've replaced it with a link to the "Layouts and Templates" page in the User Guide on docs.borg. However, that's a fragile thing to do, since that page's URL might change in the future, so I solicit alternative suggestions.

**Comment by jenlampton** [comment]
> The link to the "online documentation" previously just went to a listing of menu items. I've replaced it with a link to the "Layouts and Templates" page in the User Guide on docs.borg. However, that's a fragile thing to do, since that page's URL might change in the future, so I solicit alternative suggestions.

That's the right thing to do. If the URL changes in the future a redirect will be created.

**Comment by stpaultim** [comment]
This needs ad advocate to be added to the 1.21 milestone. However, I think that this would be considered a UX improvement and could be added to the bugfix milestone. Not sure it matters, because I think we should be able to get this into either. 

I like it and would mark it works for me, but I think it would be better to get a little more feedback on the appearance and content first. 

I still have questions about how the way we are handling this help text relates to how we are or will handle it elsewhere in core. I will bring up these questions at the Design/UX meeting tomorrow.

**Comment by bugfolder** [comment]
Discussion at the 12/2 design meeting: the word "Help" as the title might be less useful than something more specific: "How Layouts Work", or something like that. Suggestions solicited.

**Comment by klonos** [comment]
>I still have questions about how the way we are handling this help text relates to how we are or will handle it elsewhere in core.

I also don't want to hold this back, so I wouldn't worry too much about styling and consistency of these help elements - it can be a follow-up issue.

Re the concerns that were surfaced during the recent design meeting around how "prominent" this help element is, and how it may "draw attention" away from the main actions in the page, here's a quick mockup of how that would look as a "help bar", right bellow the admin bar (with reduced font size + topic subject added next to the word "Help"):

![image](https://user-images.githubusercontent.com/2423362/144667545-8bdba31d-f0b0-4cbf-8c62-ce5be50977a3.png)

That would mimic the WP paradigm a bit closer, and get this help "out of the way".

**Comment by indigoxela** [comment]
> ... get this help "out of the way".

I'm not sure if I'd realize, that this far away collapsed thing directly beneath the admin bar is the help for layouts. Things above the tabs are usually something unrelated. At least, for me.

IMO, if that details element is collapsed it's not too prominent. One thing's true: only the word "Help" isn't enough for the summary. "Help: How layouts work" seems better.

Maybe in Backdrop we don't need lots of these extra help texts, but for layouts this additional information is for sure useful. For people coming from Drupal and people coming from any other CMS.
Many things in Backdrop are sort of self-explanatory - layouts are not (views aren't, either).

What's possibly not ideal, is the current formatting of the help text - I find these nested lists hard to read.

Just my 2c. :wink:

**Comment by bugfolder** [comment]
Per comments above, I've made two changes to the current PR:

* Made the summary text "How layouts work". (I don't think that's less clear than "Help: how layouts work" and it's more direct.)
* Put the class "details-help" on the summary text and used that to restore its font size to 100% (in Seven), so it's less overwhelming.

Results in the Tugboat build; looks like this:

![sample](https://user-images.githubusercontent.com/29214084/145602062-539236c0-569c-4557-8800-b86b874eb3c0.jpg)

**Comment by jenlampton** [comment]
I love, love, love this!

Suggestions for text changes (mostly shortening / simplifying for clarity):
* `To determine which layout will be used for a page, the path is compared to those shown here.` 
* `Paths used by the layout system are <em>system paths</em> (also called normal paths), not URL aliases.` -- because there is nothing in Backdrop called a "path alias", it was renamed to "URL alias".  (We could link URL alias to `admin/config/urls/path`?)
* `Paths used by the layout system are listed alphabetically below. All layouts that might apply to each path are listed below the path.`
* `All pages without a layout explicitly defined for its path will use one of the Default layouts (shown at the bottom of this page). `
* `Each path listed here will have one or more layouts defined:`
    * `If any path has exactly one layout defined, then that layout will be used (if visibility conditions allow).`
        * `Starting from top to bottom, all layouts for that path are checked for matching visibility conditions.`
        * `The first layout that matches all conditions will be the one that gets used.`
        * `To adjust priority, you may reorder the layouts for a provided path.`
        * `If no match is found, the page will use one of the Default layouts.`

I tried to remove any mention of "table" or "group" because how the things are organized on the page isn't as important as what that organization means, but if you think it was important to mention I could be convinced otherwise :)

**Comment by bugfolder** [comment]
>  because there is nothing in Backdrop called a "path alias", it was renamed to "URL alias". 

There is a "path alias" here: https://docs.backdropcms.org/api/backdrop/core%21includes%21path.inc/function/backdrop_get_normal_path/1

This page uses both "URL alias" and "path alias":
https://docs.backdropcms.org/api/backdrop/core%21includes%21path.inc/1

A search through core code turns up 61 "path alias" but 187 "URL alias". So I can see the latter is preferred, but I guess we got some updating to do. 

Up until now, I'd been using "system path" and "normal path" to mean different things: `node/%/edit` is a system path, but `node/1/edit` is a normal path (what gets returned by `backdrop_get_normal_path()`. But on further examination of docs and functions, I see "system path" being used for "normal paths", so I guess I'm going to need to rewrite my brain-config.

So what kind of path would we call `node/%/edit` when we're trying to make distinctions between the different types of path? "Menu router path" seems a bit unwieldy and obscure for our intended audience.

**Comment by bugfolder** [comment]
Anyhow...I've made all those changes (with a few minor tweaks), and also changed the `details` class to `description` (so it will be styled by the CSS added in https://github.com/backdrop/backdrop-issues/issues/5399). Here's what it looks like (with the older styling).

![From Clipboard](https://user-images.githubusercontent.com/29214084/146481868-933a05ef-24a2-4d96-8b7c-ac51a2bae672.jpg)

**Comment by stpaultim** [comment]
Works for me! 

But, given all the text and discussion - I think I'd like one more person to endorse this version before marking it as such.

**Comment by indigoxela** [comment]
It also works for me, so marking it as such.

**Comment by klonos** [comment]
This is a task, so it could in theory get in with any bug fix release, but since the milestone label was set for the next minor, I've set the milestone to 1.21.0. If we happen to miss that window for whatever reason, it should get in with any next 1.21.x release I think.

I've reviewed the code, and I'm happy with it, so RTBC 👍🏼 

We can tweak the wording further as we see fit, and as more PRs that affect the layout listing page and/or the way layouts work get merged.

**Comment by klonos** [comment]
...re wording, I've left a couple of suggestions, but I'd prefer it if we discussed it in person, to avoid back and forth changes. It will also help if another person takes a look at this and chimes in. @jenlampton ?

**Comment by jenlampton** [comment]
I think the PR looks really good and is already a huge improvement over what we have now. PR is RTBC from me -- and any further text changes could be done post-merge if necessary.

That said, I'm going to file a PR against @bugfolder's PR (mostly so I can see what it looks like with my suggestions).

**Comment by jenlampton** [comment]
I think I have something I like. Feedback wanted! Here's where I landed:
<img width="1201" alt="Screen Shot 2022-01-02 at 8 24 23 PM" src="https://user-images.githubusercontent.com/397895/147899618-aad1835a-8b3e-4708-b1de-e17812c460d1.png">

Changes include:
* Started with a short paragraph (2 sentences) that explain the whole process of layout selection, start to end.
* Follow that with bullets describing how the page is organized.
* Moved disambiguation of "Path" to the very bottom
* Clarified what a `system path` vs `normal path` vs `URL alias` is -- with examples.
* Emphasized that normal paths should not be used (they do work right now, but that's a bug) ;)

**Comment by bugfolder** [comment]
Agreed, this is great. Jen's PR merged into https://github.com/backdrop/backdrop/pull/3754.

**Comment by docwilmot** [comment]
I have a few comments and recommendations:

- As a page is rendered, its Path is compared to those listed here to determine if a particular layout should be assigned to that Path.
- Visibility conditions will also be evaluated to determine if the user is allowed to access this layout.
- If no layout matches the Path, or visibility conditions are not met, the page will use one of the Default layouts shown at the bottom of this page.
- Each Path listed below may have one or more available layouts assigned. For paths with multiple layouts:
  - Starting from top to bottom, all layouts for that path are checked for matching visibility conditions.
  - The first layout that matches all conditions will be the one that gets used.
  - You may reorder the layouts for a given path to adjust priority.
- The layout system uses system paths, not normal paths or URL aliases.
- For more information and details, see the online documentation.

I think we should state first that paths lead to a layout, then introduce VCs after this is clear, otherwise multiple new concepts in one bullet point is too much. The path examples re system vs alias etc are more likely to confuse IMO. We know what those are but to a newbie, there is no difference between `node/1`, `about` and `node/%`; theyre all web addresses. If the user wants to clarify this, they should research the more extensive document.

**Comment by bugfolder** [comment]
Although...I just noticed that @jenlampton is referring to what layouts accept as "system paths". I think what they want are called "menu router paths" in the few places they are explicitly named, like https://docs.backdropcms.org/api/backdrop/core%21modules%21field_ui%21field_ui.module/1). 

There are many places in the documentation where the term "system path" is used as a synonym for "normal path", like:

* https://docs.backdropcms.org/api/backdrop/core%21includes%21path.inc/function/backdrop_lookup_path/1
* https://docs.backdropcms.org/api/backdrop/core%21includes%21path.inc/1
* https://docs.backdropcms.org/documentation/url-aliases
* https://docs.backdropcms.org/api/backdrop/core%21modules%21path%21path.api.php/function/hook_path_update/1
* https://docs.backdropcms.org/api/backdrop/core%21includes%21menu.inc/group/menu/1

On this page, though, "system path" is indeed used for layout paths:

* https://docs.backdropcms.org/documentation/deep-dive-building-with-blocks-and-layouts

It seems like there's more usages of "system path" as a synonym for "normal path".

On the other hand, for purposes of user-friendly documentation, "system path" is a nicer name than "menu router path", so there is some logic to adopting "system path" for that usage. But that would mean that a bunch of places in the documentation would need to be updated.

So: should "system path" be the official public name for menu router paths like `node/%`?

**Comment by docwilmot** [comment]
I think system path is a proper usage, but my comment above was more what would be understandable rather than what is accurate.

**Comment by stpaultim** [comment]
I've reviewed the text and I think that this will be very helpful to users.

**Comment by jenlampton** [comment]
> Although...I just noticed that @jenlampton is referring to what layouts accept as "system paths". I think what they want are called "menu router paths"

I'd be fine with changing that laguage to "Menu router paths", we could even add something like `As defined in hook_menu` if that would be useful.

**Comment by jenlampton** [comment]
Thoughts on @docwilmot's recommendations :)

> I think we should state first that paths lead to a layout, then introduce VCs after this is clear, otherwise multiple new concepts in one bullet point is too much.

My goal was to include all of the new concepts in a concise paragraph that painted a complete picture, before we started with the bullets.  Keeping to that model, we could do something like the following for the first sentence. 

`As a page is rendered, a Layout is selected. The Path for the page is compared to those listed here...`

**Question: Do people prefer all bullets to a paragraph before the bullets?** I went back and forth on this, and ended up deciding on a paragraph first so that people would get the whole picture of "How layouts work" before the details of "what's on the page", and "how to use it" (things in the bullets). I don't know if this separation makes sense to everyone else :)

> As a page is rendered, its Path is compared to those listed here to determine if...

I think people need to understand that a layout is *always* selected for a page. The "to determine if" makes it sound like it's possible for no layout to be selected. we could try "to determine which" (but I still feel that the current paragraph text, with shorter sentences, is more clear)

> ...a particular layout should be assigned to that Path.

I'm not comfortable with "assigned" here -- is this intended to mean "available" or "selected"? (see below)

> Each Path listed below may have one or more available layouts assigned.

I don't like "assigned" here, that sounds like they have already been selected (perhaps because of above usage?). What about just switching "one or more available layouts" to "one or more layouts available"?

> Visibility conditions will also be evaluated to determine if the user is allowed to access this layout.

This isn't quite right, because visibility conditions are not usually about user access (and, we don't want to use the word "User" anyway). "Visibility conditions will also be evaluated to determine if a layout should be selected" is more accurate.

> to a newbie, there is no difference between node/1, about and node/%; theyre all web addresses. If the user wants to clarify this, they should research the more extensive document.

I think this is even more reason that we need to provide examples. I see so many people entering `node/1`  into layouts when what they meant was `node/%`. If show them right here that `node/1` is not what they want, it will save them a lot of headaches. 

I'm going to try a PR to fix the first sentence, and switch "System path" to "Menu router path", and try to add in the word "selected" for clarification.

**Comment by jenlampton** [comment]
Okay, I think this is an improvement: 
<img width="1262" alt="Screen Shot 2022-01-03 at 1 13 30 PM" src="https://user-images.githubusercontent.com/397895/147980995-77da9eaf-aa85-4f75-8f03-e9f281c7ce7a.png">

Changes include:
* First sentence now `As a page is rendered, a Layout is selected.`
* last bullet now `Paths used by the layout system are menu router paths`
* Fixed all tense in the first paragraph is present (not a mix of future)
* Multiple layouts for a single path now called `available layout`
* Layout that is going to be used now called `selected layout`

**Comment by bugfolder** [comment]
LGTM.👍

**Comment by stpaultim** [comment]
> I'd be fine with changing that laguage to "Menu router paths", we could even add something like As defined in hook_menu if that would be useful.

I have to say that "menu router paths" actually does not mean anything to me, although I know it should. System path was easier for me to understand. HOWEVER, I think that either work fine along with the examples and I am not advocating that we need to switch back. Overall this is hugely helpful and the use of examples and "Menu router path" might even help me better understand this term which I do hear frequently.

**Comment by bugfolder** [comment]
As a follow-up, https://docs.backdropcms.org/documentation/deep-dive-building-with-blocks-and-layouts should probably be edited to use the same terminology.

(Actually, that page has some incorrect information on it: an "override layout" is not the same thing as a "dynamic layout". The former is a layout for an existing path; the latter contains a placeholder. A layout could be either (`node` or `foo/%`), both (`node/%`), or neither (`foo`).)

And perhaps we should try to standardize terminology for the various types of path throughout code and docs in a follow-up issue? (As we recently did with "URL Aliases".)

**Comment by docwilmot** [comment]
Oh dear, menu router paths is even less friendly to me. Perhaps I should ask for clarification as to who this help text is for; I believe it should at least target the people least familiar with such terminology as menu router paths.
> My goal was to include all of the new concepts in a concise paragraph that painted a complete picture, before we started with the bullets.

I dont see that as necessary, the fieldset already lends context to the idea that this would be instructions on "how layouts work", and bullet points are easier to follow.

I dont feel the point re layouts being alphabetical lends anything much to this, which is why I removed it.

 I would still combine "Each path may have one or more layouts available." and "Available layouts are listed beneath the path." into one sentence such as "Each Path listed below may have one or more available layouts assigned. For paths with multiple layouts:" because the second sentence is only relevant if the first is true.

> Paths used by the layout system are menu router paths (example: node/%), not normal paths (example: node/2) or URL aliases (example: about)

My point here is that as a first introduction to layouts, this is confusing, not the least because it could be inaccurate. Yes you *can* actually create a layout with a path that *looks like* `node/2` or `officer/25` or `about` or `cheese`. Those examples may be misleading.

**Comment by jenlampton** [comment]
> I dont feel the point re layouts being alphabetical lends anything much to this, which is why I removed it.

I added it specifically because some people were confused about how things were ordered on this page (and having it helped them) -- we can remove it if I was mistaken about that :) @stpaultim do you remember this?

> I would still combine "Each path may have one or more layouts available." and "Available layouts are listed beneath the path." into one sentence 

I had considered combining these into the same bullet also, but I don't like combining the sentences.  Let's start with the single bullet and see if that's enough of an improvement. 

**Question: Do we need the indentation in the bullets?**  I'm wondering if that adds any value. 

> My point here is that as a first introduction to layouts, this is confusing

Yes, this is a very confusing part of layouts. (We have a bunch of other issues to help with the confusion too). Unfortunately, it's also critical for people to understand what should be used for a path. 

The likelihood of someone using `node/2` when they should have used `node/%` but misunderstood, is about 99.9%.  That's more than enough reason to add a disambiguation to the first introduction to layouts. 

> Yes you can actually create a layout with a path that looks like node/2

You are correct -- but as I said above -- but that is a bug. The layout system was never intended to be used that way so it probably should not be encouraged :)

**Comment by bugfolder** [comment]
> You are correct -- but as I said above -- but that is a bug. The layout system was never intended to be used that way so it probably should not be encouraged :)

And I have created https://github.com/backdrop/backdrop-issues/issues/5433 where we can address what we might do about that to discourage it in the future without breaking things from the past. (I do like the warning message that pops up on the layout page if you try it.)

**Comment by docwilmot** [comment]
Side note, its not a bug. Backdrop recognizes that youre about to do this, and throws a warning; if you persist, then you intended to do this, and its your site, and you have your reasons, thats not a bug.

But point is those examples are misleading! If I have 3 types of cheese and I make a layout for `cheese/3` why are these instructions telling me I cannot create a layout that looks like `cheese/3`? This is what your example is suggesting: dont make layouts that look like `xxxx/number`. Even worse, dont make layouts that look like a word (about). I feel this will confuse.

**Comment by jenlampton** [comment]
> Side note, its not a bug. Backdrop recognizes that youre about to do this, and throws a warning

We added the warning because we were shocked when we discovered it was even possible (since it was not supposed to be possible -- hence the bug). 

> This is what your example is suggesting: dont make layouts that look like xxxx/number. 

That's correct. This is also what the warning is suggesting. This is still good advice. If you are an expert and want to accept the risk and do it anyway, thats fine. These instructions aren't for you :) 

Back to formatting question, here's what it might look like without indentation: is this better? 
<img width="1258" alt="Screen Shot 2022-01-03 at 3 20 43 PM" src="https://user-images.githubusercontent.com/397895/147990955-864d2d64-3c75-4187-82a8-8b45771cf957.png">

**Comment by bugfolder** [comment]
How about for the first sentence:

As a page is rendered, a _Layout_ is selected as follows. The _Path_ for the page...

Just to make it clear that the two sentences are not describing sequential things.

**Comment by jenlampton** [comment]
Here's an example with indented bullets - and the first paragraph back in bullets, closer to how it was earlier today: 
<img width="1256" alt="Screen Shot 2022-01-03 at 3 38 05 PM" src="https://user-images.githubusercontent.com/397895/147991839-b4d31e5d-8d34-4218-928e-3613fbe07eee.png">

**Comment by bugfolder** [comment]
Nice, but nitpick: the first sentence isn't part of the list, so shouldn't be bulleted.

**Comment by docwilmot** [comment]
> That's correct. This is also what the warning is suggesting. This is still good advice. If you are an expert and want to accept the risk and do it anyway, thats fine. These instructions aren't for you :)

Last try, just in case I am not being clear. The only time it is "wrong" to do this ( meaning paths that look like `xxxx/number`) is for entity paths, like node/1, user/2. To be clear: `story/2022` is ok, `cheese/3` is ok. Your example would make it seem (perhaps only to me) that a xxxx/number path is bad, because a new user doesnt know that the "node" part of a "normal path" like "node/2" is a special thing.

**Comment by docwilmot** [comment]
P.s. apart from this last point, I am happy with the last iteration.

**Comment by jenlampton** [comment]
> Your example would make it seem (perhaps only to me) that a xxxx/number path is bad, because a new user doesnt know that the "node" part of a "normal path" like "node/2" is a special thing.

Lightbulb! I get what you were saying now. Thank you for saying it 3 times.

hm...  In the example all references were for the about page. I wonder if there is a clearer way to indicate that?

**Comment by jenlampton** [comment]
Okay, here's another one with the first/last bullets removed, more indented bullet added (not sure if this is better), and another sentence attempting to clarify that the examples are all for a single page:

<img width="1269" alt="Screen Shot 2022-01-03 at 3 49 27 PM" src="https://user-images.githubusercontent.com/397895/147992445-029c6f48-2299-41ce-8d27-6f815468b757.png">

**Comment by bugfolder** [comment]
👍. One more teeny thing: in the last sentence, should it be "For the default About page,...", just to make it clear that this isn't a requirement for any site that happens to have an About page (like a site migrated from D7)?

**Comment by jenlampton** [comment]
Okay, here's a PR:  https://github.com/bugfolder/backdrop/pull/6

**Comment by bugfolder** [comment]
👍. Merged into current PR.

**Comment by stpaultim** [comment]
> I added it specifically because some people were confused about how things were ordered on this page (and having it helped them) -- we can remove it if I was mistaken about that :) @stpaultim do you remember this?

For the record, I do believe that my own misunderstanding of the order in which layouts were evaluated was part of the reason we went down this path. I was falsely under the impression that all layouts were evaluated from top to bottom (and I'm sure I've passed along this mistaken impression in recorded presentations). 

I think the new text is very helpful in rectifying this.

**Comment by stpaultim** [comment]
I think that we're all getting a bit tired and have looked at this for too long. I think we've missed a couple of grammatical errors.

1) "The Path for the page is compared those in the alphabetical list here ..." should be "The Path for the page is compared to those in the alphabetical list here ..."

2) "The top available layout is is evaluated ..." should be "The top available layout for each path is evaluated ..."

**Comment by bugfolder** [comment]
Agreed. Made.

**Comment by olafgrabienski** [comment]
Sorry for being late here. I've not followed the whole discussion but mainly skimmed the discussion about Layout system paths, and I have a question: Should the help text also be valid for paths of core layouts? These don't look like menu router paths to me.

- Dashboard: `admin/dashboard`
- Home page: `home`

**Comment by bugfolder** [comment]
> These don't look like menu router paths to me.

They are indeed menu router paths, which are created by the layouts themselves. Put differently, they are layout-created pages, versus overridden pages. (The changes to the layout listing page in https://github.com/backdrop/backdrop-issues/issues/5077 are attempting to make that distinction clearer.)

**Comment by olafgrabienski** [comment]
> They are indeed menu router paths, which are created by the layouts themselves.

Thanks for the clarification! They *look* more like URL aliases, but I guess it's difficult to cover these cases here.

**Comment by jenlampton** [comment]
this conversation is making me realize how confusing everything is. I'm so impressed that all of you managed to figure out layouts, and grateful that you are all here to help others ❤️

**Comment by jenlampton** [comment]
This PR does not need a thorough code review because it's mostly adding interface text, but I can't do it anyway because I was heavily involved in updates to the PR myself. If I could, I'd mark it RTBC :D Is there anyone else available to do so?

**Comment by stpaultim** [comment]
@jenlampton - As simple as this is, I'm not quite comfortable marking it RTBC. However, I did leave one note/question.

**Comment by indigoxela** [comment]
This issue proves, how hard wording can be. :wink:

Great result! RTBC :+1:

**Comment by irinaz** [comment]
Looks good, I think there might be more feedback from users after this help text goes live :)

**Comment by quicksketch** [comment]
Out of habit I merged this into 1.x and 1.20.x, as it doesn't contain any new features. So I'm moving this to the 1.20.4 milestone.

I think we can work on the presentation and overall approach to extended help text in a follow-up issue. I wouldn't want all the effort to write this text go to waste or wait, and this gives us a good example to work from in the future. Thanks everyone!
